### PR TITLE
Chore: updated authentication, roles and user signup 

### PR DESCRIPTION
**Pull request by Ajinx1**
All test cases passed

Screenshot 2024-07-23 at 10 50 35 PM
Screenshot 2024-07-23 at 10 53 23 PM
Screenshot 2024-07-23 at 10 54 02 PM
Screenshot 2024-07-23 at 10 54 19 PM

### CODE REVIEW

File: internal/models/user.go [hunk 1]
```diff
@@ -65,3 +67,18 @@ func (u *User) CreateUser(db *gorm.DB) error {
 
 	return nil
 }
+
+func (u *User) GetSeedUsers(db *gorm.DB) ([]User, error) {
+	var users []User
+
+	if err := db.Preload("Profile").Preload("Products").Preload("Organisations").Limit(2).Find(&users).Error; err != nil {
+		return users, err
+	}
```
Comment (NwokoyeChigozie) [inline internal/models/user.go]: create a generic function in repository that can help with these multiple preloads

File: pkg/controller/user/user.go [hunk 2]
```diff
@@ -20,75 +16,38 @@ type Controller struct {
...
-	}
+	// // perform get user request here
 
-	base.Logger.Info("user login successfully")
+	// base.Logger.Info("user created successfully")
+	// rd := utility.BuildSuccessResponse(http.StatusCreated, "user created successfully",)
 
-	rd := utility.BuildSuccessResponse(http.StatusOK, "user login successfully", respData)
-	c.JSON(http.StatusOK, rd)
+	// c.JSON(http.StatusOK, rd)
```
Comment (NwokoyeChigozie) [inline pkg/controller/user/user.go]: are you still using these?

**Review by NwokoyeChigozie — CHANGES_REQUESTED** [review]

File: pkg/controller/user/user.go (hunk 2 above)
Comment (Ajinx1) [reply pkg/controller/user/user.go]: not yet sir, fixed.

File: internal/models/user.go (hunk 1 above)
Comment (Ajinx1) [reply internal/models/user.go]: done sir

**Pull request closed without merging**